docs/migrate-matomo-4-to-5.md

```diff
@@ -0,0 +1,128 @@
+
+### Logging
+
+We used to use `\Monolog\Logger` or `\Psr\Log\LoggerInterface` to access our logging interface through dependency injection.
+* Use `\Piwik\Log\Logger` instead of `\Monolog\Logger`
+* Use `\Piwik\Log\LoggerInterface` instead of `\Psr\Log\LoggerInterface`
+* Use `\Piwik\Log\NullLogger` instead of `\Psr\Log\NullLogger`
+
+To access our logging interface you can e.g. use
+```php
+$logger = \Piwik\Container\StaticContainer::get(\Piwik\Log\Logger::class);
+```
+
+Or when injection the logger into e.g. an API constructor using dependency injection you can use
+```php
+public function __construct(\Piwik\Log\LoggerInterface $logger)
+{
+    $this->logger = $logger;
+```
+
+### Console Commands
+
+Our console commands were using parts of the symfony console directly. Command classes will no longer be allowed to any symfony components diretly, instead we have rewritten our base class for commands `\Piwik\Plugins\ConsoleCommand`. This class will no give you access to console functionalities directly. To update your plugin command you need to apply the following changes:
+* Methods like `run`, `execute`, `interact` or `initialize` can no longer be overwritten. Instead, use our custom methods prefixed with `do`: `doExecute`, `doInteract` or `doInitialize`
+  * `doExecute()` method needs to return integers. We recommend using the class constants `SUCCESS` or `FAILURE` as return values.
+* Where ever you need to work with input or output streams use `$this->getInput()` or `$this->getOutput` instead. Don't use `InputInterface` or `OutputInterface` as method typehints.
+* When defining input options and arguments `addOption` and `addArgument` can no longer be used
+  * For arguments use `addOptionalArgument` or `addRequiredArgument`
+  * For options use `addNegatableOption`, `addOptionalValueOption`, `addNoValueOption` or `addRequiredValueOption`
+* Directly using any console helpers is now prohibited
+  * To ask for user input use the new methods `askForConfirmation`, `askAndValidate` or `ask`
+  * For a progress bar use the methods `initProgressBar`, `startProgressBar`, `advanceProgressBar` and `finishProgressBar`
+  * Tables can be rendered using the new method `renderTable`
+* For executing another command within your command use the new method `runCommand`
+
+## Tests on CI
+
+We used to use Travis CI for testing. With Matomo 5 we discontinued support for running tests on Travis CI, instead it is possible to use a GitHub action for this.
+You can find more details on how to set up your own GitHub test action in [this guide](/guides/tests-github).
+
+## JQuery Updated
+
+jQuery has been updated to 3.6.3. Please check your plugin's JavaScript code if it needs to be adjusted. More details can be found in jQuery update guides: https://jquery.com/upgrade-guide/3.0/ and https://jquery.com/upgrade-guide/3.5/
+
+## Plugin names
```

Comment:
Does renaming a plugin have any other implications we may need to mention here? Would it  e.g. appear as a new plugin or is there some automatic connection between the two different names?

Reply:
At least for plugins published on the marketplace this doesn't make a difference. The marketplace was allowing only 40 characters before, so such plugins won't be affected. So this actually only affects custom plugins someone might have running locally. Renaming the plugin might require to update the config options, database settings and similar stuff if the plugin has any.